jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpReceiverOverHTTP3.java

```diff
@@ -71,82 +94,31 @@ public void onResponse(Stream.Client stream, HeadersFrame frame)
         MetaData.Response response = (MetaData.Response)frame.getMetaData();
         httpResponse.version(response.getHttpVersion()).status(response.getStatus()).reason(response.getReason());
 
-        if (responseBegin(exchange))
+        responseBegin(exchange);
```

Comment:
Check for `exchange.isResponseComplete()` here.

Reply:
Done.